apps/web/src/pages/ApplicationFormPage/ApplicationFormPage.tsx

```diff
@@ -0,0 +1,427 @@
+          <label
+            className={classes.marginBottom30px}
+            htmlFor="highSchoolOrCollegeName"
+          >
+            Naziv srednje škole ili fakulteta koji trenutno pohađaš:
+          </label>
+          <Input
+            placeholder="Your answer"
+            type="text"
+            id="highSchoolOrCollegeName"
+            {...register('highSchoolOrCollegeName')}
+          />
+        </div>
+        <div className={classes.formQuestionWrapper}>
+          <label className={classes.marginBottom30px} htmlFor="channel">
+            Za DUMP internship si saznao/la putem:
+          </label>
+
+          <FormControl>
+            <RadioGroup defaultValue="medium" name="radio-buttons-group">
+              <Radio
+                value={FoundOutAboutInternshipBy.Friend}
+                label="Prijatelja ili poznanika"
+                {...register('foundOutAboutInternshipBy', {
+                  required: 'This field is required',
+                })}
+              />
+              <Radio
+                value={FoundOutAboutInternshipBy.SocialMedia}
+                label="Društvenih mreža"
+                {...register('foundOutAboutInternshipBy')}
+              />
+              <Radio
+                value={FoundOutAboutInternshipBy.Presentation}
+                label="Predstavljanja na fakultetima/školama"
+                {...register('foundOutAboutInternshipBy')}
+              />
+              <Radio
+                value={FoundOutAboutInternshipBy.Media}
+                label="Medija"
+                {...register('foundOutAboutInternshipBy')}
+              />
+              <Radio
+                value={FoundOutAboutInternshipBy.Other}
+                label="Ostalo"
+                {...register('foundOutAboutInternshipBy')}
+              />
+            </RadioGroup>
+          </FormControl>
+
+          {errors.foundOutAboutInternshipBy ? (
+            <p className={classes.warningText}>
+              {errors.foundOutAboutInternshipBy?.message}
+            </p>
+          ) : (
+            <div className={classes.warningTextPlaceholder}></div>
+          )}
+        </div>
+        <div className={classes.formQuestionWrapper}>
+          <label
+            className={classes.marginBottom30px}
+            htmlFor="reasonForApplying"
+          >
+            Zašto se prijavljuješ na DUMP internship?
+          </label>
+          <Input
+            placeholder="Your answer"
+            type="text"
+            id="firstName"
+            {...register('reasonForApplying', {
+              required: 'Reason for applying is required',
+            })}
+          />
+
+          {errors.reasonForApplying ? (
+            <p className={classes.warningText}>
+              {errors.reasonForApplying?.message}
+            </p>
+          ) : (
+            <div className={classes.warningTextPlaceholder}></div>
+          )}
+        </div>
```

Comment:
Općenito u cijelom ovom dijelu loše je što je "zahardkodiran" pitanjima, pa ako želiš dodati jedno, moraš na nekoliko mjesta updateati. Mislim da bi bilo dobro imati pitanja (bar ova iz data) spremljena u array, gdje je svako pitanje objekt poput {type: QuestionType.Radio, label: 'Kako si čuo', validation: {required: true}, options: ['a', 'b']}  ili nešto tako. Na temelju toga izračunaš default vrijednost forme da se ne bi input mijenjao iz uncontrolled u controlled, a pitanja mapiraš i imaš komponentu koja će vratiti input field na temelju questiona koji proslijediš. Ja sam zapravo to napravio već u svom PR-u pa možda možeš to iskoristiti, procijeni sam kako će biti najbolje.

Reply:
Also, materialui nudi dosta komponenti i opcija za izgled svega, pogotovo formi, pa možeš pogledati neki njihov primjer i ovdje implementirati umjesto da se mučiš klasama i css-om 👀